src/Confluent.SchemaRegistry/SchemaRegistryConfig.cs

```diff
@@ -14,12 +14,30 @@
 //
 // Refer to LICENSE for more information.
 
+using System;
 using System.Collections;
 using System.Collections.Generic;
 
 
 namespace Confluent.SchemaRegistry
 {
+    /// <summary>
+    ///     Auth credentials source.
+    /// </summary>
+    public enum AuthCredentialsSource
+    {
+        /// <summary>
+        ///     Credentials are specified via the `schema.registry.basic.auth.user.info` config property in the form username:password.
+        ///     If `schema.registry.basic.auth.user.info` is not set, authentication is disabled.
+        /// </summary>
+        UserInfo,
+
+        /// <summary>
+        ///     Credentials are specified via the `sasl.username` and `sasl.password` configuration properties.
+        /// </summary>
+        SaslInherit
+    }
```

Comment:
what about retrieving it from the URL?

Reply:
left as a todo. I had problems getting this to work and just omitted it in the end.